Automattic/jetpack
# External Media: Add arrow keys navigation

Adds arrow key navigation on media items in the external media browser modal. This brings it to have parity with the core Media Library image grid navigation.

![arrow-key-media-grid](https://user-images.githubusercontent.com/967608/84435242-1a049080-abf7-11ea-9e93-0ebd863a8729.gif)

#### Changes proposed in this Pull Request:
* Adds a `onKeyDown` prop to `MediaItem`
* Adds a `columns` ref and window resize listener to count and store the number of grid image columns. This is used to know where to send focus to when the up/down arrow keys are pressed.

#### Testing instructions:






* Go to a post and add an Image or Gallery block
* **Arrow key presses _should not close the modal_**. Focus different elements (search input, search button, etc) and press the arrow keys. It should never close the modal. Arrow keys should operate normally inside inputs (like the search input).
* **Media Grid image items should be able to navigate with arrow key presses**
* **Resize the browser to less than or greater than 600px so the images restack.** The `columns` number should be updated and up/down arrow keys should bring you to the correct image.
* **All other functionality should still work** (Selecting images via click, space, enter; load more, etc)
